distributed/protocol/compression.py

```diff
@@ -93,6 +93,39 @@ def _fixed_lz4_decompress(data):
     }
     default_compression = "lz4"
 
+with ignoring(ImportError):
+    _import_policy = dask.config.get("distributed.comm.zstd.import_policy")
+    os.environ["PYTHON_ZSTANDARD_IMPORT_POLICY"] = _import_policy
+
+    import zstandard as zstd
+
+    zstd_level = dask.config.get("distributed.comm.zstd.level")
+    zstd_kwargs = dict(
+        write_checksum=dask.config.get("distributed.comm.zstd.write_checksum"),
+        write_content_size=dask.config.get("distributed.comm.zstd.write_content_size"),
+        write_dict_id=dask.config.get("distributed.comm.zstd.write_dict_id"),
+        threads=dask.config.get("distributed.comm.zstd.threads"),
+    )
+
+    if zstd_level is not None:
+        zstd_kwargs["level"] = zstd_level
+
+    zstd_compressor = zstd.ZstdCompressor(**zstd_kwargs)
+    zstd_decompressor = zstd.ZstdDecompressor()
+
+    def zstd_compress(data):
+        if isinstance(data, (memoryview, bytearray)):
+            data = bytes(data)
+        return zstd_compressor.compress(data)
+
+    def zstd_decompress(data):
+        if isinstance(data, (memoryview, bytearray)):
+            data = bytes(data)
```

Comment:
I recommend either `distributed.utils.ensure_bytes` or just call `bytes(data)` in all cases (remove the `if`).  This will also support other types that support the buffer protocol.  It is also cheap.

```python
In [1]: data = b'0' * 1000000

In [2]: %timeit bytes(data)
91.4 ns ± 2.16 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```

Reply:
commit#47d05a5,   zstandard from Conda has "cffi" import policy